composer/datasets/streaming/dataset.py

```diff
@@ -96,10 +97,19 @@ def __init__(self,
         self.timeout = timeout
         self.batch_size = batch_size
 
+        compression_basename = get_compression_scheme_basename()
+        compression_local = self._download_file(compression_basename, wait=(dist.get_local_rank() != 0))
+        with open(compression_local, 'r+') as fp:
+            compression_scheme = fp.read()
+            self.compression_scheme = compression_scheme if compression_scheme != '' else None
```

Comment:
Curious why are we truncating the file?

Reply:
local datasets have no compression scheme, so we empty the compression metadata file.

Reply:
Would it work to leave the file as-is if local?

Reply:
you're right, we can't support compression for `remote==local` anyway, I think we should throw some kind of error in `__init__` if the user attempts this.